### PR TITLE
[TOOLS-2212] Comments are being fetched from Jira and showed on the card details

### PR DESCRIPTION
**Pull request by emannuelalmeida**

### CODE REVIEW

File: cardboard/src/main/java/objective/taskboard/controller/CardDto.java [hunk 1]
```diff
@@ -9,13 +9,8 @@
...
 import com.google.common.base.Objects;
 
-import objective.taskboard.data.Changelog;
-import objective.taskboard.data.CustomField;
-import objective.taskboard.data.Issue;
+import objective.taskboard.data.*;
 import objective.taskboard.data.Issue.CardTeam;
-import objective.taskboard.data.Subtask;
-import objective.taskboard.data.TaskboardTimeTracking;
-import objective.taskboard.data.User;
```
Comment (mauricio-frontend) [inline cardboard/src/main/java/objective/taskboard/controller/CardDto.java]: I noticed that is many classes at _taskboard/data_ folder. My doubt is if import all classes doesn't reflect at performance?

File: core/src/main/java/objective/taskboard/domain/converter/JiraIssueToIssueConverter.java [hunk 2]
```diff
@@ -47,14 +47,12 @@
...
+import objective.taskboard.data.*;
+import objective.taskboard.jira.client.JiraCommentDto;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
 import objective.taskboard.cycletime.CycleTime;
-import objective.taskboard.data.Issue;
-import objective.taskboard.data.IssueScratch;
-import objective.taskboard.data.TaskboardTimeTracking;
-import objective.taskboard.data.User;
```
Comment (mauricio-frontend) [inline core/src/main/java/objective/taskboard/domain/converter/JiraIssueToIssueConverter.java]: Some doubt hear, isn't better import only the Class that will be used?

File: core/src/test/java/objective/taskboard/data/IssueTest.java [hunk 3]
```diff
@@ -8,13 +8,7 @@
 import static org.mockito.Mockito.when;
 
 import java.lang.reflect.Field;
-import java.util.Arrays;
-import java.util.Date;
-import java.util.LinkedHashMap;
-import java.util.LinkedList;
-import java.util.List;
-import java.util.Optional;
-import java.util.Set;
```
Comment (mauricio-frontend) [inline core/src/test/java/objective/taskboard/data/IssueTest.java]: The same doubt about import is applied hear.

File: core/src/main/java/objective/taskboard/data/IssueScratch.java [hunk 4]
```diff
@@ -47,6 +47,8 @@
     protected CustomField classOfService;
     @JsonIgnore
     protected List<Worklog> worklogs; //NOSONAR
+
+    protected List<Comment> comments;
```
Comment (objective-solutions-user) [inline core/src/main/java/objective/taskboard/data/IssueScratch.java]: CRITICAL Make "comments" private or transient. [rule](http://10.44.1.250:9000/coding_rules#rule_key=squid%3AS1948)

**Comment by objective-solutions-user** [comment]
SonarQube analysis reported 1 issue
* ![CRITICAL][CRITICAL] 1 critical

Watch the comments in this conversation to review them.

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'

File: cardboard/src/main/resources/static/elements/taskboard/taskboard-home.html [hunk 5]
```diff
@@ -576,6 +576,7 @@
                             taskboard.setLaneConfiguration(data.laneConfiguration);
                             taskboard.urlJira = data.urlJira;
                             taskboard.urlLinkGraph = data.urlLinkGraph;
+                            taskboard._setBallparks(data.ballparks);
```
Comment (mauricio-frontend) [inline cardboard/src/main/resources/static/elements/taskboard/taskboard-home.html]: Please remove this line, it is no longer used.

File: core/src/main/resources/static/scripts/taskboard.js [hunk 6]
```diff
@@ -601,6 +601,32 @@ function Taskboard() {
             .value();
     };
 
+    this._setBallparks = function(data){
```
Comment (mauricio-frontend) [inline core/src/main/resources/static/scripts/taskboard.js]: Please remove this method, it is no longer used.

File: core/src/main/resources/static/scripts/taskboard.js [hunk 7]
```diff
@@ -601,6 +601,32 @@ function Taskboard() {
...
+        this.ballparks = data;
+    };
+
+    this._getBallparks = function(type) {
```
Comment (mauricio-frontend) [inline core/src/main/resources/static/scripts/taskboard.js]: Please remove this method, it is no longer used.

File: cardboard/src/main/resources/static/elements/taskboard/issue-detail-comments.html [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+                margin-left: 8px;
+                margin-right: 8px;
+                word-break: break-all;
+            }
+
+        </style>
+
+        <div id="comments-content" class="comments--panel">
+            <template is="dom-repeat" items="{{item.comments}}" as="comment">
+                <paper-material class="comment-panel" elevation="0">
```
Comment (jhonypg) [inline cardboard/src/main/resources/static/elements/taskboard/issue-detail-comments.html]: Pode trocar para esse paper-material. Quando não há a necessidade de usar um componente como esse, é melhor evitar para não pegar códigos aleatórios.

File: cardboard/src/main/resources/static/elements/taskboard/issue-detail-comments.html [hunk 9]
```diff
@@ -0,0 +1,124 @@
...
+                        notify: true
+                    },
+                    selected: {
+                        type: Number,
+                        value: () => 0
+                    }
+                },
+
+                _formatCreatedDate: (created) =>  {
+
```
Comment (jhonypg) [inline cardboard/src/main/resources/static/elements/taskboard/issue-detail-comments.html]: Remover essa linha pulada. O mesmo no método _formatCommentBody.

File: cardboard/src/main/resources/static/elements/taskboard/issue-detail-comments.html [hunk 10]
```diff
@@ -0,0 +1,124 @@
...
+    </template>
+
+    <script>
+        (function () {
+            Polymer({
+                is: 'issue-detail-comments',
+
+                properties: {
+                    item: {
+                        observer: '_itemUpdated',
```
Comment (jhonypg) [inline cardboard/src/main/resources/static/elements/taskboard/issue-detail-comments.html]: Como esse _itemUpdated não existe, pode remover esse observer.

File: cardboard/src/main/resources/static/elements/taskboard/issue-detail-comments.html [hunk 11]
```diff
@@ -0,0 +1,124 @@
...
+            }
+
+        </style>
+
+        <div id="comments-content" class="comments--panel">
+            <template is="dom-repeat" items="{{item.comments}}" as="comment">
+                <paper-material class="comment-panel" elevation="0">
+                    <div class="comment-header">
+                        <div class="comment-author">
+                            <img class="comment-avatar" src$="{{comment.avatar}}" title$="{{comment.author}}"/>
```
Comment (jhonypg) [inline cardboard/src/main/resources/static/elements/taskboard/issue-detail-comments.html]: Sempre que não houver two way databinding ou que ele não for necessário, usar "[[  xxx ]]" no lugar de "{{ xxx }}".

File: cardboard/src/main/resources/static/elements/taskboard/issue-detail-comments.html [hunk 12]
```diff
@@ -0,0 +1,124 @@
...
+                            <h5 class="author-name">{{comment.author}}</h5>
+                        </div>
+                        <div class="comment-time-since">
+                            <span class="">
```
Comment (jhonypg) [inline cardboard/src/main/resources/static/elements/taskboard/issue-detail-comments.html]: Como essa tag span não é necessária, pode remover.

File: cardboard/src/main/resources/static/elements/taskboard/ballpark-item.html [hunk 13]
```diff
@@ -48,7 +48,7 @@
              .size-value-wrapper {
                 position:relative;
              }
-             
+
```
Comment (jhonypg) [inline cardboard/src/main/resources/static/elements/taskboard/ballpark-item.html]: Reverter.

File: cardboard/src/main/resources/static/elements/taskboard/issue-detail-comments.html [hunk 14]
```diff
@@ -0,0 +1,124 @@
+<dom-module id="issue-detail-comments">
+    <template>
+        <style is="custom-property">
+
+            :host {
+                display: block;
+                width: 100%;
+            }
+
+            paper-tab {
```
Comment (jhonypg) [inline cardboard/src/main/resources/static/elements/taskboard/issue-detail-comments.html]: Parece que esse seletor não é usado.

File: cardboard/src/main/resources/static/elements/taskboard/issue-detail-comments.html (hunk 8 above)
Comment (jhonypg) [reply cardboard/src/main/resources/static/elements/taskboard/issue-detail-comments.html]: Nesse caso, poderia remover também o elevation="0".

File: cardboard/src/main/resources/static/elements/taskboard/issue-detail-comments.html [hunk 15]
```diff
@@ -0,0 +1,124 @@
...
+
+            marked-element {
+                margin-left: 8px;
+                margin-right: 8px;
+                word-break: break-all;
+            }
+
+        </style>
+
+        <div id="comments-content" class="comments--panel">
```
Comment (jhonypg) [inline cardboard/src/main/resources/static/elements/taskboard/issue-detail-comments.html]: Deixa assim: ```<div class="comments-content">```

File: cardboard/src/main/resources/static/elements/taskboard/issue-detail-comments.html [hunk 16]
```diff
@@ -0,0 +1,124 @@
...
+            .comment-author {
+                display: flex;
+                align-items: center;
+            }
+
+            .comment-avatar {
+                width: 32px;
+                height: 32px;
+                overflow: hidden;
+                background-color: #ccc;
```
Comment (jhonypg) [inline cardboard/src/main/resources/static/elements/taskboard/issue-detail-comments.html]: Adiciona: "color: #ccc;"

File: cardboard/src/main/resources/static/elements/taskboard/issue-detail-comments.html (hunk 11 above)
Comment (jhonypg) [reply cardboard/src/main/resources/static/elements/taskboard/issue-detail-comments.html]: Adiciona alt$="[[comment.author]]"

File: cardboard/src/main/resources/static/elements/taskboard/issue-detail-comments.html [hunk 17]
```diff
@@ -0,0 +1,124 @@
...
+                is: 'issue-detail-comments',
+
+                properties: {
+                    item: {
+                        observer: '_itemUpdated',
+                        notify: true
+                    },
+                    selected: {
+                        type: Number,
+                        value: () => 0
```
Comment (jhonypg) [inline cardboard/src/main/resources/static/elements/taskboard/issue-detail-comments.html]: Mudar para ```value: 0```

File: cardboard/src/main/resources/static/elements/taskboard/issue-detail-comments.html [hunk 18]
```diff
@@ -0,0 +1,124 @@
+<dom-module id="issue-detail-comments">
+    <template>
+        <style is="custom-property">
```
Comment (jhonypg) [inline cardboard/src/main/resources/static/elements/taskboard/issue-detail-comments.html]: Pode remover esse is="custom-property"

File: core/src/main/java/objective/taskboard/domain/converter/IssueFieldsExtractor.java [hunk 19]
```diff
@@ -20,42 +20,27 @@
...
-import objective.taskboard.jira.client.JiraCommentDto;
-import objective.taskboard.jira.client.JiraComponentDto;
-import objective.taskboard.jira.client.JiraFieldDataDto;
-import objective.taskboard.jira.client.JiraIssueDto;
-import objective.taskboard.jira.client.JiraIssueLinkTypeDto;
-import objective.taskboard.jira.client.JiraLinkDto;
-import objective.taskboard.jira.client.JiraWorklogResultSetDto;
-import objective.taskboard.jira.properties.JiraProperties;
-import objective.taskboard.utils.DateTimeUtils;
+import java.util.*;
```
Comment (objective-solutions-user) [inline core/src/main/java/objective/taskboard/domain/converter/IssueFieldsExtractor.java]: CRITICAL Explicitly import the specific classes needed. [rule](http://10.44.1.250:9000/coding_rules#rule_key=squid%3AS2208)

File: core/src/main/java/objective/taskboard/domain/converter/IssueFieldsExtractor.java [hunk 20]
```diff
@@ -20,42 +20,27 @@
...
+import java.util.stream.*;
```
Comment (objective-solutions-user) [inline core/src/main/java/objective/taskboard/domain/converter/IssueFieldsExtractor.java]: CRITICAL Explicitly import the specific classes needed. [rule](http://10.44.1.250:9000/coding_rules#rule_key=squid%3AS2208)

File: core/src/main/java/objective/taskboard/domain/converter/IssueFieldsExtractor.java [hunk 21]
```diff
@@ -20,42 +20,27 @@
...
+
+import org.codehaus.jettison.json.*;
```
Comment (objective-solutions-user) [inline core/src/main/java/objective/taskboard/domain/converter/IssueFieldsExtractor.java]: CRITICAL Explicitly import the specific classes needed. [rule](http://10.44.1.250:9000/coding_rules#rule_key=squid%3AS2208)

File: core/src/main/java/objective/taskboard/domain/converter/IssueFieldsExtractor.java [hunk 22]
```diff
@@ -20,42 +20,27 @@
...
+
+import objective.taskboard.data.*;
```
Comment (objective-solutions-user) [inline core/src/main/java/objective/taskboard/domain/converter/IssueFieldsExtractor.java]: CRITICAL Explicitly import the specific classes needed. [rule](http://10.44.1.250:9000/coding_rules#rule_key=squid%3AS2208)

File: core/src/main/java/objective/taskboard/domain/converter/IssueFieldsExtractor.java [hunk 23]
```diff
@@ -20,42 +20,27 @@
...
+import objective.taskboard.jira.*;
```
Comment (objective-solutions-user) [inline core/src/main/java/objective/taskboard/domain/converter/IssueFieldsExtractor.java]: CRITICAL Explicitly import the specific classes needed. [rule](http://10.44.1.250:9000/coding_rules#rule_key=squid%3AS2208)

File: core/src/main/java/objective/taskboard/domain/converter/IssueFieldsExtractor.java [hunk 24]
```diff
@@ -20,42 +20,27 @@
...
+import objective.taskboard.jira.client.*;
```
Comment (objective-solutions-user) [inline core/src/main/java/objective/taskboard/domain/converter/IssueFieldsExtractor.java]: CRITICAL Explicitly import the specific classes needed. [rule](http://10.44.1.250:9000/coding_rules#rule_key=squid%3AS2208)

File: core/src/main/java/objective/taskboard/domain/converter/IssueFieldsExtractor.java [hunk 25]
```diff
@@ -20,42 +20,27 @@
...
+import objective.taskboard.jira.properties.*;
```
Comment (objective-solutions-user) [inline core/src/main/java/objective/taskboard/domain/converter/IssueFieldsExtractor.java]: CRITICAL Explicitly import the specific classes needed. [rule](http://10.44.1.250:9000/coding_rules#rule_key=squid%3AS2208)

File: core/src/main/java/objective/taskboard/domain/converter/IssueFieldsExtractor.java [hunk 26]
```diff
@@ -20,42 +20,27 @@
...
+import objective.taskboard.utils.*;
```
Comment (objective-solutions-user) [inline core/src/main/java/objective/taskboard/domain/converter/IssueFieldsExtractor.java]: CRITICAL Explicitly import the specific classes needed. [rule](http://10.44.1.250:9000/coding_rules#rule_key=squid%3AS2208)

**Comment by objective-solutions-user** [comment]
SonarQube analysis reported 8 issues
* ![CRITICAL][CRITICAL] 8 critical

Watch the comments in this conversation to review them.

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'

**Pull request merged**